From 46efbd29914c0ea1174b6fe2e47461d8a1f5e0ed Mon Sep 17 00:00:00 2001
From: Sean Tranchetti <stranche@codeaurora.org>
Date: Thu, 22 Mar 2018 16:49:38 -0600
Subject: [PATCH 05/10] sepolicy: Add rps to sysfs context and allow netmgrd
 access

Netmgrd accesses /sys/devices/virtual/net/rmnet_*/queues/rx-0/rps_cpus,
so we must use proper labeling of the /sys/devices/virtual/net
directory.

Definitions for various other sysfs contexts have been updated to
allow for consistency with the other branches.

Change-Id: I04e83082df9deaad9b772b01c6434e52467a3f51
CRs-Fixed: 2199494
---
 common/genfs_contexts | 2 ++
 common/netmgrd.te     | 5 +++++
 2 files changed, 7 insertions(+)

diff --git a/common/genfs_contexts b/common/genfs_contexts
index cac3200..fd80b61 100644
--- a/common/genfs_contexts
+++ b/common/genfs_contexts
@@ -19,3 +19,5 @@ genfscon sysfs /kernel/wcd_cpe0 u:object_r:sysfs_audio:s0
 genfscon sysfs /devices/virtual/thermal u:object_r:sysfs_thermal:s0
 genfscon sysfs /devices/virtual/kgsl/kgsl/proc u:object_r:sysfs_kgsl_proc:s0
 genfscon debugfs /kgsl/proc     u:object_r:kgsl_debugfs:s0
+genfscon sysfs /module/tcp_cubic/parameters u:object_r:sysfs_net:s0
+genfscon sysfs /devices/virtual/net u:object_r:sysfs_net:s0
diff --git a/common/netmgrd.te b/common/netmgrd.te
index 714f541..ed35b01 100644
--- a/common/netmgrd.te
+++ b/common/netmgrd.te
@@ -98,3 +98,8 @@ get_prop(netmgrd, hwservicemanager_prop)
 hwbinder_use(netmgrd)
 binder_call(netmgrd, netd)
 allow netmgrd system_net_netd_hwservice:hwservice_manager find;
+
+#Allow diag logging
+userdebug_or_eng(`
+  r_dir_file(netmgrd, sysfs_diag)
+')
-- 
2.17.1

